prog/dftb+/lib_io/hsdparser.F90

```diff
@@ -616,6 +618,9 @@ function createChildNode(parentNode, childName, curLine, file) result(newChild)
           dummy => removeChild(parentNode, sameChild)
           call destroyNode(sameChild)
           tCreate = .true.
+        elseif (iType == 6) then
+          newChild => null()
+          return
         else
```

Comment:
Case 6 should be handled separately. There is now reason to look up for other nodes with the same name, if you find a comment sign in front of a node. What about:
```
if (iType == 0) then
...
else if (iType == 6) then
  newChild => null()
  return
else
...
end if
```